src/workerd/io/worker-entrypoint.c++

```diff
@@ -399,6 +403,10 @@ kj::Promise<WorkerInterface::AlarmResult> WorkerEntrypoint::runAlarm(
           [scheduledTime, entrypointName=entrypointName, &context,
            &metrics = incomingRequest->getMetrics()]
           (Worker::Lock& lock){
+        // TODO(cleanup): Here we rely on the root level JS being evaluated synchronously in
+        // ServiceWorkerGlobalScope::request().  Is there a more reliable way to express this?
+        jsg::AsyncContextFrame::StorageScope scope = context.makeAsyncTraceScope(lock);
```

Comment:
I see you've only covered HTTP and alarms. Shouldn't we cover all the event types? Or is the assumption here that this only matters for events that can be delivered to actors? Even so, I think we expect TCP streams should be supported by actors and other event types could be added as well, so I'd try to support them all.

Reply:
From what I could tell, the remaining handlers to cover were `connect()`, which is currently `KJ_UNIMPLEMENTED`, and `runScheduled()` and `customEvent()`, which I wasn't sure could be covered by the same assumption about synchronous JS evaluation as I was making about `request()`.  But I didn't know about the method for capturing and reentering async frames that James mentioned.  I can try implementing those, too.

I assume that `WorkerEntrypoint` is still the right place to implement the async frame capture, and not, say, in the `ServiceWorkerGlobalScope` or `CustomEvent` methods that it calls.